pkg/cloud/aws/services/userdata/node.go

```diff
@@ -39,6 +39,30 @@ nodeRegistration:
 EOF
 
 kubeadm join --config /tmp/kubeadm-node.yaml
+`
+
+	nodeCloudInit = `{{.Header}}
+write_files:
+-   path: /tmp/kubeadm-node.yaml
+    owner: root:root
+    permissions: '0644'
```

Comment:
Since this file contains secrets it might be better to set permissions to `0640` instead

Reply:
Good call, I changed it.